client-course-schedulizer/src/utilities/helpers/caseFunctions.ts

```diff
@@ -24,175 +24,227 @@ const thursReg = RegExp("[Tt][Hh]|[Rr]");
+    return SemesterLength.HalfFirst;
+  }
+  if (lowerValue === "second") {
+    return SemesterLength.HalfSecond;
+  }
+  if (lowerValue === "full") {
+    return SemesterLength.Full;
+  }
+  if (upperValue === "A") {
+    return SemesterLength.IntensiveA;
+  }
+  if (upperValue === "B") {
+    return SemesterLength.IntensiveB;
+  }
+  if (upperValue === "C") {
+    return SemesterLength.IntensiveC;
+  }
+  if (upperValue === "D") {
+    return SemesterLength.IntensiveD;
+  }
+  // eslint-disable-next-line no-console
+  console.log(`Half of "${value}" is unreadable, defaulting to Full`);
+  return SemesterLength.Full;
+};
+
+export const daysCase = (value: string) => {
+  const days: Day[] = [];
   if (sunReg.test(value)) {
-    firstMeeting.days.push(di.Day.Sunday);
+    days.push(Day.Sunday);
   }
   if (monReg.test(value)) {
-    firstMeeting.days.push(di.Day.Monday);
+    days.push(Day.Monday);
   }
   if (tuesReg.test(value)) {
-    firstMeeting.days.push(di.Day.Tuesday);
+    days.push(Day.Tuesday);
   }
   if (wedReg.test(value)) {
-    firstMeeting.days.push(di.Day.Wednesday);
+    days.push(Day.Wednesday);
   }
   if (thursReg.test(value)) {
-    firstMeeting.days.push(di.Day.Thursday);
+    days.push(Day.Thursday);
   }
   if (friReg.test(value)) {
-    firstMeeting.days.push(di.Day.Friday);
+    days.push(Day.Friday);
   }
   if (satReg.test(value)) {
-    firstMeeting.days.push(di.Day.Saturday);
+    days.push(Day.Saturday);
   }
+  return days;
 };
 
-export const instructorCase = (value: string, { section }: CaseCallbackParams) => {
+export const instructorCase = (value: string): string[] => {
   const instructors = value.split(/[;,\n]/);
-  section.instructors = instructors.map((instructor) => {
+  return instructors.map((instructor) => {
     return instructor.trim();
   });
 };
 
-export const sectionStartCase = (value: string, { section }: CaseCallbackParams) => {
-  section.startSectionDate = value;
-};
-
-export const sectionEndCase = (value: string, { section }: CaseCallbackParams) => {
-  const sectionStart = moment(section.startSectionDate, "l");
+export const sectionEndCase = (
+  value: string,
+  startSectionDate: string | undefined,
+): SemesterLength => {
+  const sectionStart = moment(startSectionDate, "l");
   const sectionEnd = moment(value, "l");
   const sectionLength = sectionEnd.diff(sectionStart, "days");
   const startMonth = sectionStart.month();
   const firstStartMonths = [0, 1, 7, 8]; // Jan, Feb, Aug, Sept
   if (sectionLength > 80) {
-    section.semesterLength = di.SemesterLength.Full;
-  } else if (sectionLength > 35 && sectionLength <= 80) {
-    section.semesterLength = firstStartMonths.includes(startMonth)
-      ? di.SemesterLength.HalfFirst
-      : di.SemesterLength.HalfSecond;
-  } else {
-    // TODO: Figure out if intensive is A, B, C, or D
-    section.semesterLength = di.SemesterLength.IntensiveA;
+    return SemesterLength.Full;
   }
+  if (sectionLength > 35 && sectionLength <= 80) {
```

Comment:
It could be good to make these numbers be constants. Without the context, it is hard to derive the significance of 35 and 80. 

something like `const minHalfLength = 35;`

Reply:
My bad... I believe I made these magic numbers in a previous PR haha

Reply:
Haha oops, I missed them too.

Reply:
f14e906